lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -3588,6 +3600,62 @@ def test_build_url(self):
         url = self.base_block.build_url(base_url, params, utm_params)
         assert url == 'http://test.server?param1=value1&param2=value2&utm_source=edx.org'
 
+    @patch.dict(settings.FEATURES, {"ENABLE_MKTG_SITE": False})
+    def test_get_learn_more_button_url__marketing_not_enabled(self):
+        """
+        Test that when the marketing site isn't enabled, the learn more button
+        goes to the 'about_course' view
+        """
+        self.setup_course()
+        CourseOverviewFactory.create(
+            id=self.course.id,
+        )
+        url = self.base_block.get_learn_more_button_url(
+            self.course, {}
+        )
+        self.assertEqual(
+            url,
+            reverse('about_course', kwargs={'course_id': str(self.course.id)})
+        )
+
+    @ddt.unpack
+    @ddt.data(
+        (True, False),
+        (False, True),
+        (True, True),
+        (True, False)
+    )
+    @patch.dict(settings.FEATURES, {"ENABLE_MKTG_SITE": True})
+    def test_get_learn_more_button_url(self, has_marketing_url, has_override):
+        """
+        Test for learn more button url behavior when the marketing site is
+        enabled but considering overrides and missing marketing urls.
+        """
+        self.setup_course()
+        utm_params = {'utm_source': 'askjeeves'}
+        overview = CourseOverviewFactory.create(id=self.course.id)
+        if has_marketing_url:
+            overview.marketing_url = 'mysite.com/this_course'
```

Comment:
I couldn't get this to work just passing it into the factory create method? I don't know why.